Document recoverability pipeline (#5632)
Particular/docs.particular.net: nservicebus/recoverability/pipeline.md

---
title: Recoverability pipeline
summary: Extending the recoverability pipeline with custom behaviors
reviewed: 2022-02-10
component: Core
versions: '[8.0,)'
---

> The recoverability [pipeline](/nervicebus/pipeline) allows advanced customization of metadata captured during message failures and also full control over the recoverability action taken.

Author: ```suggestion
The recoverability [pipeline](/nervicebus/pipeline) allows for advanced customization of the metadata captured during message failures as well as full control over the recoverability action taken.
```

Reviewer: Applied manually

The example below shows how to [register a behavior](/nservicebus/pipeline/manipulate-with-behaviors.md#add-a-new-step) that:

> - Stores the message body separately
> - Excludes it from the message sent to the error queue
> - Adds a metadata entry that links to the stored body

Author: ```suggestion
- Stores the message body in an external storage
- Excludes the body from the message sent to the error queue
- Adds a metadata entry that links to the stored body
```
